### PR TITLE
Add commands to Speedrun cog for random run records

### PR DESCRIPTION
**Pull request by jakeHebert**
Okay, so there are a few things going on here. 

I changed around the Speedrun cog a bit so that I could add the new commands to it and make everything fit in well. I created a command group called speedrun and changed the command for finding a random video from just ```.speedrun``` to ```.speedrun video```.

I used the Python wrapper for the Speedrun.com API. It is fairly slow, taking about 5-6 seconds to pull and parse the data to send to Discord, and while I didn't test the efficiency, I don't think that using the Rest API was faster.

As far as I can tell, the Speedrun.com API requires that you use the 'id' to find whatever genre, platform, etc, so I manaully scraped the site for all the ids for platform and genre (around 1300 ids) and added them to a JSON file so that we can grab the id by matching the string to it.

Lastly, I was playing around with adding another bot message with the video linked to the run after the embed, but I thought it was a bit too much clutter per command.

Attached is an image of the commands for platform and genre.
speedrunCommands

### CODE REVIEW

File: bot/seasons/evergreen/speedrun.py [hunk 1]
```diff
@@ -1,28 +1,136 @@
 import json
 import logging
+import re
 from pathlib import Path
 from random import choice
 
+import srcomapi
```
[the remark this answers is not in the document]
Comment (jakeHebert) [reply bot/seasons/evergreen/speedrun.py]: I ran ```pipenv install srcomapi``` and let it do its thing. Do I need to do anything else?

**Comment by sco1** [comment]
I have concerns about using this API as-is. Since it's using `requests` to query the API, it blocks the bot for a not-insignificant amount of time when making API calls. `.speedrun year 2018`, for example, takes me about 10 seconds to run.

Because the asynchronous API calls are buried pretty intimiately into the data models that the API it's not a quick migration to asyncio. Ultimately, I think the effort is better spent making our own wrapper for the API. Is this something you'd like to give a try? If not, no worries, we can work on getting something put together so we can get this command ready to go.

**Comment by jakeHebert** [comment]
@sco1 I would love to try and figure this out, although I am not very familiar with async in Python. I'll have to spend some time doing some research. 
I'd assume a good place to start would be asyncio and aiohttp? 
If there is anyone who could give some advice or help I would greatly appreciate it.

**Comment by sco1** [comment]
The [`aiohttp` Client Quickstart](https://aiohttp.readthedocs.io/en/stable/client_quickstart.html) has pretty much everything you'd need to replace the equivalent `requests` code. Replacing `requests` is probably one of the easier things on the list, the rest of the time is going to be spent generating the appropriate data models & methods to achieve the desired behavior.

**Comment by jakeHebert** [comment]
@sco1 I abandoned the idea of using/creating a wrapper class for this, because I don't think I'll have the time to invest into it at the moment. 
So I went ahead and used the REST API and made all the requests async. It cut the time of the commands down by about 50% most of the time. On my machine, most commands are satisfied within 3.5 seconds.
I also added some Exception handling in as well to catch any weird things that happen (like games not having any speedrun records).

**Comment by sco1** [comment]
@jakeHebert please use branches if you're going to be working on this repository, every commit you're pushing to your `master` branch is triggering a CI build.

**Comment by jakeHebert** [comment]
@sco1 Sorry about that, I was trying to fix the issues the checks were bringing up.

**Pull request closed without merging**